compiler/next/include/chpl/resolution/resolution-types.h

```diff
@@ -20,45 +20,71 @@
 #ifndef CHPL_RESOLUTION_RESOLUTION_TYPES_H
 #define CHPL_RESOLUTION_RESOLUTION_TYPES_H
 
-#include "chpl/uast/TypeDecl.h"
+#include "chpl/types/Type.h"
+#include "chpl/uast/ASTNode.h"
 
 namespace chpl {
 namespace resolution {
 
 
-class Type {
- public:
-  enum TypeTag {
-    FROM_DECL,
-    INT8,
-    INT16,
-    INT32,
-    INT64,
-    UINT8,
-    UINT16,
-    UINT32,
-    UINT64,
-    BOOL,
-    BOOL8,
-    BOOL16,
-    BOOL32,
-    BOOL64,
-    REAL32,
-    REAL64,
-    IMAG32,
-    IMAG64,
-    COMPLEX64,
-    COMPLEX128,
-  };
- private:
-  TypeTag tag_;
-  const uast::TypeDecl* typeDecl_;
- public:
-  TypeTag tag() { return tag_; } 
-  const uast::TypeDecl* typeDecl() { return typeDecl_; }
-  UniqueString name(Context* context);
+struct ResolutionResult {
```

Comment:
Could we consider naming this `ResolvedExpression`? I think it would parallel `ResolvedSymbol`. Alternatively something like `ResolvedUse` and `ResolvedDef`. Right now `ResolutionResult` leaves me confused about the purpose of `ResolvedSymbol`.

Reply:
Sure, but I'll name it ResolvedExpression in a follow-up work. This section here is really just some stuff unrelated to the PR that I'm leaving alone to avoid creating merge conflicts for myself.